crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -0,0 +1,539 @@
+            0x0009_0068 => Ok(ScardIoCtlCode::ForgetReaderA),
+            0x0009_006C => Ok(ScardIoCtlCode::ForgetReaderW),
+            0x0009_0070 => Ok(ScardIoCtlCode::AddReaderToGroupA),
+            0x0009_0074 => Ok(ScardIoCtlCode::AddReaderToGroupW),
+            0x0009_0078 => Ok(ScardIoCtlCode::RemoveReaderFromGroupA),
+            0x0009_007C => Ok(ScardIoCtlCode::RemoveReaderFromGroupW),
+            0x0009_0098 => Ok(ScardIoCtlCode::LocateCardsA),
+            0x0009_009C => Ok(ScardIoCtlCode::LocateCardsW),
+            0x0009_00A0 => Ok(ScardIoCtlCode::GetStatusChangeA),
+            0x0009_00A4 => Ok(ScardIoCtlCode::GetStatusChangeW),
+            0x0009_00A8 => Ok(ScardIoCtlCode::Cancel),
+            0x0009_00AC => Ok(ScardIoCtlCode::ConnectA),
+            0x0009_00B0 => Ok(ScardIoCtlCode::ConnectW),
+            0x0009_00B4 => Ok(ScardIoCtlCode::Reconnect),
+            0x0009_00B8 => Ok(ScardIoCtlCode::Disconnect),
+            0x0009_00BC => Ok(ScardIoCtlCode::BeginTransaction),
+            0x0009_00C0 => Ok(ScardIoCtlCode::EndTransaction),
+            0x0009_00C4 => Ok(ScardIoCtlCode::State),
+            0x0009_00C8 => Ok(ScardIoCtlCode::StatusA),
+            0x0009_00CC => Ok(ScardIoCtlCode::StatusW),
+            0x0009_00D0 => Ok(ScardIoCtlCode::Transmit),
+            0x0009_00D4 => Ok(ScardIoCtlCode::Control),
+            0x0009_00D8 => Ok(ScardIoCtlCode::GetAttrib),
+            0x0009_00DC => Ok(ScardIoCtlCode::SetAttrib),
+            0x0009_00E0 => Ok(ScardIoCtlCode::AccessStartedEvent),
+            0x0009_00E4 => Ok(ScardIoCtlCode::ReleaseTartedEvent),
+            0x0009_00E8 => Ok(ScardIoCtlCode::LocateCardsByAtrA),
+            0x0009_00EC => Ok(ScardIoCtlCode::LocateCardsByAtrW),
+            0x0009_00F0 => Ok(ScardIoCtlCode::ReadCacheA),
+            0x0009_00F4 => Ok(ScardIoCtlCode::ReadCacheW),
+            0x0009_00F8 => Ok(ScardIoCtlCode::WriteCacheA),
+            0x0009_00FC => Ok(ScardIoCtlCode::WriteCacheW),
+            0x0009_0100 => Ok(ScardIoCtlCode::GetTransmitCount),
+            0x0009_0104 => Ok(ScardIoCtlCode::GetReaderIcon),
+            0x0009_0108 => Ok(ScardIoCtlCode::GetDeviceTypeId),
+            _ => {
+                error!("Unsupported ScardIoCtlCode: 0x{:08x}", value);
+                Err(invalid_message_err!("try_from", "ScardIoCtlCode", "unsupported value"))
+            }
+        }
+    }
+}
+
+/// Allow [`ScardIoCtlCode`] to be used as an [`IoCtlCode`].
+impl IoCtlCode for ScardIoCtlCode {}
+
+/// [2.2.2.30 ScardAccessStartedEvent_Call]
+///
+/// [2.2.2.30 ScardAccessStartedEvent_Call]: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpesc/c5ab8dd0-4914-4355-960c-0a527971ea69
+#[derive(Debug)]
+pub struct ScardAccessStartedEventCall {
+    _unused: u32,
+}
+
+impl ScardAccessStartedEventCall {
+    pub fn decode(payload: &mut ReadCursor<'_>) -> PduResult<Self> {
+        Ok(Self {
+            _unused: payload.read_u32(),
+        })
+    }
+}
```

Comment:
Nit: should this be ignored and discarded like we do for padding?
```suggestion
#[derive(Debug)]
pub struct ScardAccessStartedEventCall;

impl ScardAccessStartedEventCall {
    pub fn decode(payload: &mut ReadCursor<'_>) -> PduResult<Self> {
        ironrdp_pdu::read_padding!(payload, 4);
        Ok(Self)
    }
}
```
(equivalent to `ironrdp_pdu::padding::read(payload, 4)`)

Reply:
8571c44c501f6efdf1406530cb279008e421ef86